### PR TITLE
Make dangerous shuttles unbuyable + emags & hacking

### PR DESCRIPTION
**Pull request by Sinestia**
## About The Pull Request

Here's all it does: 
A. Dangerous shuttles are no longer available for purchase by default, these are the arena, disco shuttle, oh hi daniel, hyperfractal, and the giant meteor.
B. However, the comms console can be emagged to re-enable these. Emag also anonymizes your name (it did this before but i don't think people knew)
C. In addition, you can use a screwdriver on a communications console circuit board to re-enable them as well. Antags without emags can thus order them, and captains with crew approval/intense circumstances can order them as well. 
D. The ID of the one who authorized the purchase is printed in the chat. So if you use a captain's ID, it will look like they ordered it. 
E. Dangerous shuttles are presented in their own category at the bottom, and text has been altered to indicate these changes.

 I based this off of Archanial's PR #2841 & PowerfulBacon's recommendation. So thanks to them. 

Closes #2828. 
## Why It's Good For The Game

Dangerous shuttles are banbait generally. This makes it so that you have to be much more deliberate in calling them, while also opening the door for hijackers and other antags to keep them from being dead content. 

## Changelog
:cl:Froststahr
add: Emags allow traitors to anonymously purchase any shuttle they want without ID! Screwdrivers can be used on circuit boards as well, but adequate access is still required in that case.
tweak: The name of the authorizer, not the buyer, is printed on purchase of a shuttle. 
tweak: Unsafe shuttles are now unavailable for purchase without emags/hacking. 
/:cl:

### CODE REVIEW

**Comment by Sinestia** [comment]
oh im also rubbish at coding and stuff so if I've done anything blatantly wrong/off feel free to tell me. Otherwise it tests fine for me.

File: code/game/objects/items/circuitboards/computer_circuitboards.dm [hunk 1]
```diff
@@ -47,8 +47,22 @@
...
 	name = "Communications (Computer Board)"
 	icon_state = "command"
+	desc = "Can be modified using a screwdriver."
 	build_path = /obj/machinery/computer/communications
 	var/lastTimeUsed = 0
+	var/insecure = 0 // Forbids shuttles that are set as illegal. 
+
+/obj/item/circuitboard/computer/communications/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		insecure = (insecure == 0) ? 1 : 0
```
Comment (PowerfulBacon) [inline code/game/objects/items/circuitboards/computer_circuitboards.dm]:
```suggestion
		insecure = !insecure
```

File: code/game/objects/items/circuitboards/computer_circuitboards.dm [hunk 2]
```diff
@@ -47,8 +47,22 @@
...
 	icon_state = "command"
+	desc = "Can be modified using a screwdriver."
 	build_path = /obj/machinery/computer/communications
 	var/lastTimeUsed = 0
+	var/insecure = 0 // Forbids shuttles that are set as illegal. 
+
+/obj/item/circuitboard/computer/communications/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		insecure = (insecure == 0) ? 1 : 0 
+		if(insecure == 1)
```
Comment (PowerfulBacon) [inline code/game/objects/items/circuitboards/computer_circuitboards.dm]:
```suggestion
		if(insecure)
```

**Comment by super12pl** [comment]
Hacking comms console takes literally 40 seconds and requires screwdriver, wirecutter and crowbar. This doesn't stop griefers from buying dangerous shuttles at all.
Remove hacking functionality but keep emag one.

**Comment by PowerfulBacon** [comment]
The original intent of the PR was to prevent people from accidentally buying dangerous shuttles. If griefers want to buy dangerous shuttles they still can, get banned and then have an admin immediately reverse it by pressing 1 button. This removes the excuse from people being 'it was accident' or they didn't know and makes it so you can only do it by grief making administrating it much easier.

**Comment by Sinestia** [comment]
Other antags too. 

I also swear I can remember admins saying that it's acceptable with crew support. Like if some miner kills bubblegum and wants to do the arena and few people object then why not pretty much. **Especially** on golden

**Review by PowerfulBacon — DISMISSED** [review]

**Review by Crossedfall — APPROVED** [review]
Config ready to go. 👍

File: code/controllers/subsystem/mapping.dm [hunk 3]
```diff
@@ -386,7 +386,8 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 			space_ruins_templates[R.name] = R
 
 /datum/controller/subsystem/mapping/proc/preloadShuttleTemplates()
-	var/list/unbuyable = generateMapList("[global.config.directory]/unbuyableshuttles.txt")
+	var/list/unbuyable = generateMapList("[global.config.directory]/shuttles_unbuyable.txt")
```
Comment (Crossedfall) [inline code/controllers/subsystem/mapping.dm:389]: This needs to be reflected in the repo. The old config file should be renamed accordingly.

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Take back my prior approval. Noticed that a config file was renamed in the code, but wasn't updated in the repo.

File: code/controllers/subsystem/mapping.dm (hunk 3 above)
Comment (Sinestia) [reply code/controllers/subsystem/mapping.dm:389]: Clarify? I'm not sure what this means, my apologies. Do you mean update the CL?

Comment (PowerfulBacon) [reply code/controllers/subsystem/mapping.dm:389]: You renamed the file, but not the filepath in code to the file

Comment (PowerfulBacon) [reply code/controllers/subsystem/mapping.dm:389]: Wait, if I am not mistaken the file names are the same

Comment (Sinestia) [reply code/controllers/subsystem/mapping.dm:389]: Yes, when I do a search in vscode I cannot find any references to unbuyableshuttles.txt, and the old file doesn't exist. 

Maybe the history appears weird from basing this off of someone elses PR i dunno.

Comment (Crossedfall) [reply code/controllers/subsystem/mapping.dm:389]: https://github.com/BeeStation/BeeStation-Hornet/blob/master/config/unbuyableshuttles.txt
https://github.com/BeeStation/BeeStation-Hornet/blob/master/config/Sage/unbuyableshuttles.txt

Are the files you need to rename.

Comment (Sinestia) [reply code/controllers/subsystem/mapping.dm:389]: Those files do not exist in this branch

Comment (Crossedfall) [reply code/controllers/subsystem/mapping.dm:389]: Pull the latest master, it should resolve the issue I'm seeing.

**Comment by Sinestia** [comment]
oh god i broke it git is weird

edit: i'll leave it for a sec til I can find some help lol

File: code/game/objects/items/circuitboards/computer_circuitboards.dm [hunk 4]
```diff
@@ -47,8 +47,22 @@
 /obj/item/circuitboard/computer/communications
 	name = "Communications (Computer Board)"
 	icon_state = "command"
+	desc = "Can be modified using a screwdriver."
 	build_path = /obj/machinery/computer/communications
 	var/lastTimeUsed = 0
+	var/insecure = 0 // Forbids shuttles that are set as illegal.
```
Comment (PowerfulBacon) [inline code/game/objects/items/circuitboards/computer_circuitboards.dm]:
```suggestion
	var/insecure = FALSE // Forbids shuttles that are set as illegal. 
```

**Comment by ike709** [comment]
Is this stale?

**Review by Crossedfall — DISMISSED** [review]
Looks good to go as far as I can tell.

File: code/game/machinery/computer/communications.dm [hunk 5]
```diff
@@ -554,10 +554,13 @@
 
 		if(STATE_PURCHASE)
 			var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
+			var/obj/item/circuitboard/computer/communications/CM = circuit
 			dat += "Budget: [D.account_balance] Credits.<BR>"
 			dat += "<BR>"
-			dat += "<b>Caution: Purchasing dangerous shuttles may lead to mutiny and/or death.</b><br>"
-			dat += "<BR>"
+			if(obj_flags & EMAGGED || CM.insecure)
```
Comment (ike709) [inline code/game/machinery/computer/communications.dm]:
```suggestion
			if((obj_flags & EMAGGED) || CM.insecure)
```

File: code/game/machinery/computer/communications.dm [hunk 6]
```diff
@@ -566,6 +569,16 @@
 					if(S.prerequisites)
 						dat += "Prerequisites: [S.prerequisites]<BR>"
 					dat += "<A href='?src=[REF(src)];operation=buyshuttle;chosen_shuttle=[REF(S)]'>(<font color=red><i>Purchase</i></font>)</A><BR><BR>"
+			if(obj_flags & EMAGGED || CM.insecure)
```
Comment (ike709) [inline code/game/machinery/computer/communications.dm]:
```suggestion
			if((obj_flags & EMAGGED) || CM.insecure)
```

**Pull request merged**